metrics-operator/controllers/analysis/objectives_evaluator.go

```diff
@@ -0,0 +1,49 @@
+package analysis
+
+import (
+	"github.com/go-logr/logr"
+	metricsapi "github.com/keptn/lifecycle-toolkit/metrics-operator/api/v1alpha3"
+	metricstypes "github.com/keptn/lifecycle-toolkit/metrics-operator/controllers/common/analysis/types"
+	"github.com/keptn/lifecycle-toolkit/metrics-operator/controllers/common/providers"
+	"golang.org/x/net/context"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+//go:generate moq -pkg fake -skip-ensure -out ./fake/evaluator_mock.go . IObjectivesEvaluator
+type IObjectivesEvaluator interface {
+	Evaluate(ctx context.Context, providerType string, obj chan metricstypes.ProviderRequest)
+}
+
+type ObjectivesEvaluator struct {
+	*metricsapi.Analysis
+	providers.ProviderFactory
+	client.Client
+	log     logr.Logger
+	results chan metricsapi.ProviderResult
+	cancel  context.CancelFunc
+}
+
+func (oe ObjectivesEvaluator) Evaluate(ctx context.Context, providerType string, obj chan metricstypes.ProviderRequest) {
```

Comment:
just checked - here we do have to keep it exported, as this function is part of the interface `IObjectivesEvaluator` that we introduced for decoupling purposes. Since the mocked implementations live in their own package, the method has to be exported to enable the unit tests to use the mocked implementation